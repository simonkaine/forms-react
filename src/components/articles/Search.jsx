import React from 'react';
import PropTypes from 'prop-types';

// build out the search and form submit function to export

export default function Search({ search, onSubmit, onChange }) {
    return (
        <form onSubmit={onSubmit}>
            <label htmlFor="search">Search</label>
                <input 
                id="search" 
                type="text" 
                name="search" 
                value={search}
                onChange={onChange}
                ></input>
            <button aria-label="button">Let's see it!</button>
        </form>
    )
}
Search.propTypes = {
    search: PropTypes.string,
    onSubmit: PropTypes.func,
    onChange: PropTypes.func
}