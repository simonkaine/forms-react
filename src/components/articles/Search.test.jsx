import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('Tests for search input', () => {
    it('Should test for search', () => {
        const search = 'discuss'
        const { asFragment } = render(<Search search={search}/>);
        expect(asFragment()).toMatchSnapshot();
    })
});
