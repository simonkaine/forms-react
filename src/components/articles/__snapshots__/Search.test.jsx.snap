// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for search input Should test for search 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search"
    >
      Search
    </label>
    <input
      id="search"
      name="search"
      type="text"
      value="discuss"
    />
    <button
      aria-label="button"
    >
      Let's see it!
    </button>
  </form>
</DocumentFragment>
`;
